### ISSUE TITLE
Measure time spent waiting for read-write locks and possibly optimize some common hot-spots (e.g. symbol table)

### ISSUE THREAD

**Issue by mkustermann** [dart-lang/sdk#45900]
We should measure how much time we spend waiting on read-write locks and possibly optimize some common hot-spots. For example symbol table lookups are on the hot path for flow graph building, so we should investigate whether we can have optimistic lock-free lookups there.

/cc @aam

**Comment by mkustermann** [comment]
Closing this in favor of https://github.com/dart-lang/sdk/issues/46252
